core/audio_processor.py

```
@@ -0,0 +1,175 @@
+                logger.info(f"Successfully processed audio: {audio.filename}")
+                return audio_data
+                
+            finally:
+                # Cleanup temporary files
+                self._cleanup_temp_files([temp_input_path, temp_output_path])
+                
+        except (AudioProcessingError, UnsupportedAudioFormatError, AudioTooLongError):
+            raise
+        except Exception as e:
+            logger.error(f"Unexpected error processing audio {audio.filename}: {e}")
+            raise AudioProcessingError(f"Failed to process audio: {str(e)}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)